collector/collector.go

```diff
@@ -167,21 +164,43 @@ func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
 func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	e.mu.Lock() // ensure no simultaneous scrapes
 	defer e.mu.Unlock()
-	if e.config.ScrapeInterval == 0 { // if we are to scrape when the request is made
-		e.scrape(ch)
-	} else {
-		scrapeResults := e.scrapeResults // There is a risk that e.scrapeResults will be replaced while we traverse this look. This should mitigate that risk
-		for idx := range scrapeResults {
-			ch <- scrapeResults[idx]
-		}
-	}
+	e.scrape(ch)
 	ch <- e.duration
 	ch <- e.totalScrapes
 	ch <- e.error
 	e.scrapeErrors.Collect(ch)
 	ch <- e.up
 }
 
+// RunScheduledScrapes is only relevant for users of this package that want to set the scrape on a timer
+// rather than letting it be per Collect call
+func (e *Exporter) RunScheduledScrapes(ctx context.Context, si time.Duration) {
+	ticker := time.NewTicker(si)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			metricCh := make(chan prometheus.Metric, 5)
+			go func() {
+				scrapeResults := []prometheus.Metric{}
```

Comment:
Should this be `e.scrapeResults`?

Otherwise, this doesn't seem to get used anywhere